zerver/tests/test_docs.py

```diff
@@ -385,6 +385,48 @@ def test_integration_404s(self) -> None:
         result = self.client_get("/integrations/doc-html/nonexistent_integration", follow=True)
         self.assertEqual(result.status_code, 404)
 
+    def test_integration_doc_images(self) -> None:
+        images_in_docs = set()
+
+        page = self.client_get("/integrations/", subdomain="").content.decode("utf-8")
+        for image in re.findall(r"/static/images/integrations/(logos/.*)\"", page):
+            images_in_docs.add(image)
+
+        for integration in INTEGRATIONS:
+            doc = self.client_get(
+                f"/integrations/doc-html/{integration}",
+                subdomain="",
+                HTTP_X_REQUESTED_WITH="XMLHttpRequest",
+            ).content.decode("utf-8")
+            for image in re.findall(r"/static/images/integrations/(.*)\"", doc):
+                images_in_docs.add(image)
```

Comment:
This logic needs to be added to the existing `test_integration_doc_endpoints`. Rendering 100 pages is expensive, so if we don't want our test suites to be really slow, we want to only have one loop over every single one of them.